StudenSchool_DO/HWALL/HW3/Provider/Courses.cs

```diff
@@ -0,0 +1,150 @@
+        while (sqlDataReader.Read())
+        {
+            result = $"{sqlDataReader["CourseID"]} {sqlDataReader["Title"]}\n";
+        }
+
+        return result;
+    }
+
+    public string GetFromDB(Guid courseID)
+    {
+        var result = string.Empty;
+
+        using var sqlConnection = new SqlConnection(ConnectionString);
+
+        var formatedSQL = string.Format(GET_SQL, courseID);
+        var sqlCommand = new SqlCommand(formatedSQL, sqlConnection);
+
+        sqlConnection.Open();
+
+        using var sqlDataReader = sqlCommand.ExecuteReader();
+        while (sqlDataReader.Read())
+        {
+            result = $"{sqlDataReader["CourseID"]} {sqlDataReader["Title"]}\n";
+        }
+
+        return result;
+    }
+
+    public void CreateFromDB(Guid courseID, string title)
+    {
+        using var sqlConnection = new SqlConnection(ConnectionString);
+
+        var formatedSQL = string.Format(CREATE_SQL, courseID.ToString(), title);
+
+        var sqlCommand = new SqlCommand(formatedSQL, sqlConnection);
+
+        sqlConnection.Open();
+
+        sqlCommand.ExecuteNonQuery();
+    }
+
+    public void EditFromDB(Guid courseID, string title)
+    {
+        using var sqlConnection = new SqlConnection(ConnectionString);
+
+        var formatedSQL = string.Format(EDIT_SQL, courseID, title);
+
+        var sqlCommand = new SqlCommand(formatedSQL, sqlConnection);
+
+        sqlConnection.Open();
+
+        sqlCommand.ExecuteNonQuery();
+    }
+
+    public void DeleteFromDB(Guid courseID)
+    {
+        using var sqlConnection = new SqlConnection(ConnectionString);
+
+        var formatedSQL = string.Format(DELETE_SQL, courseID);
+
+        var sqlCommand = new SqlCommand(formatedSQL, sqlConnection);
+
+        sqlConnection.Open();
+
+        sqlCommand.ExecuteNonQuery();
+    }
+
+    public string GetColumns()
+    {
+        return "Guid CoursesID, string Title";
+    }
+
+    public string GetAll()
+    {
+        return GetAllFromDB();
+    }
+
+    public string Get()
+    {
+        DesignedMenu.WriteServiceMessages("coursesID ");
+
+        Guid coursesID = CorrectInput.GetGuidFromUser();
+
+        return GetFromDB(coursesID);
+    }
+
+    public void Create()
+    {
+        DesignedMenu.WriteServiceMessages("coursesID ");
+
+        Guid coursesID = CorrectInput.GetGuidFromUser();
+
+        DesignedMenu.WriteServiceMessages("title ");
+
+        string title = Console.ReadLine();
+
+        CreateFromDB(coursesID, title);
+    }
+
+    public void Edit()
+    {
+        DesignedMenu.WriteServiceMessages("coursesID ");
+
+        Guid coursesID = CorrectInput.GetGuidFromUser();
+
+        DesignedMenu.WriteServiceMessages("title ");
+
+        string title = Console.ReadLine();
+
+        CreateFromDB(coursesID, title);
+    }
+
+    public void Delete()
+    {
+        DesignedMenu.WriteServiceMessages("coursesID ");
+
+        Guid coursesID = CorrectInput.GetGuidFromUser();
+
+        DeleteFromDB(coursesID);
+    }
```

Comment:
you just ignored what I was talking about on previous lesson. Repository should only get data from DB.